applications/ShapeOptimizationApplication/python_scripts/algorithm_penalized_projection.py

```diff
@@ -109,59 +118,59 @@ def RunOptimizationLoop(self):
 
     # --------------------------------------------------------------------------
     def FinalizeOptimizationLoop(self):
-        self.DataLogger.FinalizeDataLogging()
-        self.Analyzer.FinalizeAfterOptimizationLoop()
+        self.data_logger.FinalizeDataLogging()
+        self.analyzer.FinalizeAfterOptimizationLoop()
 
     # --------------------------------------------------------------------------
     def __initializeNewShape(self):
-        self.ModelPartController.UpdateMeshAccordingInputVariable(SHAPE_UPDATE)
-        self.ModelPartController.SetReferenceMeshToMesh()
+        self.model_part_controller.InitializeNewOptimizationStep(self.optimization_iteration)
```

Comment:
It seems strange that in the specific function `__initializeNewShape` a more general function `InitializeNewOptimizationStep` is called. 
I would either rename the function of the model_part_controller: `InitializeNewShape` or `InitializeModelPartUpdate`
or call it in a specific function in the algorithm: `__initializeOptimizationStep`

Reply:
You are right. I called it "UpdateTimeStep" now to better represent what is actually done. What do you think?